opentelekomcloud/services/dcs/resource_opentelekomcloud_dcs_instance_v1.go

```diff
@@ -390,6 +449,14 @@ func resourceDcsInstancesV1Create(ctx context.Context, d *schema.ResourceData, m
 		}
 	}
 
+	if _, ok := d.GetOk("whitelist"); ok {
+		whitelistOpts := getInstanceWhitelistOpts(d)
+		if err := whitelists.Put(client, d.Id(), whitelistOpts).ExtractErr(); err != nil {
+			return fmterr.Errorf("error updating redis whitelist of DCS instance: %w", err)
+		}
+		time.Sleep(5 * time.Second)
```

Comment:
try to avoid explicit waits

Reply:
The problem somehow happened because api returns the list of elements in reverse order.
Changing `whitelsit` from `TypeList` to `TypeSet` fixed the issue.

Reply:
after some test runs looks like it was not the only issue... I'm adding `StateChangeConf` func for additional check.